app/Model/DecisionThree.hs

```diff
@@ -1,48 +1,128 @@
+        choice <- getSex sex
+        choice <- getHair hair
+        choice <- getSkin skin
+        choice <- getEyes eyes
+        choice <- getProps props
+        where choice [] = []
 
 -- Todo: uma função que verifica se a persona "montada" é a mesma que a escolhida
+verificaAcertos::[String]->[String]->Bool
+verificaAcertos persona personaCorreta =
+    personaCorreta == (verificaCaracteristicas persona personaCorreta 0)
+
+-- Retorna uma nova lista com as caracteristicas da persona "montada" que estao
+-- na persona correta ou a string "erro" na ordem: (sex, hair, skin, eyes, props)
+
+verificaCaracteristicas::[String]->[String]->Int->[String]
+verificaCaracteristicas personaMontada personaCorreta i =
+    | (null caracteristica) = []  -- fim da lista / condicao de parada
+
+    | otherwise =
+        | (caracteristica  == personaCorreta!!i) =
+            [caracteristica] ++ (verificaCaracteristicas personaMontada personaCorreta i+1)
+
+        | otherwise =
+            ["erro"] ++ (verificaCaracteristicas personaMontada personaCorreta i+1)
+    where caracteristica = personaMontada!!i
 
 -- Todo: uma função que utiliza a da linha 22 e repassa para as funções da linha 11
 -- em diante que: caso o elemento retornado seja igual ao já selecionado anteriormente
 -- validar se o valor booleano do mesmo é false(ou seja, pode ser selecionado de novo)
--- se for, a função vai "rodar" novamente
-
+-- se for, a função vai "rodar" novamente.
+-- Retorna: uma lista com nomes das funcoes que ainda podem rodar
 
+funcoesRestantes::[[String]]->[String]->[String]->[(T,Boolean)]
+funcoesRestantes [categoria:caudaC] [acerto:caudaA] [funcao:caudaF] =
+    | (categoria == []) = []     -- fim da lista / condicao de parada
 
+    | acerto `elem` categoria =  -- se acertou a categoria:
+        adicionaTupla (funcao, True) (funcoesRestantes caudaC caudaA caudaF)  -- adiciona (nomeDaFuncao, True) a resposta
 
+    | otherwise =
+        adicionaTupla (funcao, False) (funcoesRestantes caudaC caudaA caudaF) -- adiciona (nomeDaFuncao, False) a resposta
 
+adicionaTupla::(T,Bool)->[T]->[T]
+adicionaTupla tupla lista = [tupla] ++ lista
 
--- Imprime reticências na tela para humanizar a tomada de decisão da IA. Perguntar ao professor ou monitor se a definição do método 
+-- Imprime reticências na tela para humanizar a tomada de decisão da IA. Perguntar ao professor ou monitor se a definição do método
 -- pode utilizar um valor concreto ao invés de um tipo
 
-esperaIA::Integer -> () -- Sempre iniciar com 0
-esperaIA inicio,totalReticencias = 
-    if inicio < 3
-        then do threadDelay 300000 -- Aprox. 0.3 segundos
-                threadDelay 300000 putStr(". ") 
-                threadDelay 300000 putStr(". ")
-                threadDelay 300000 putStrLn(".")
-                threadDelay 300000
-                esperaIA inicio+1 totalReticencias
-        else putStrLn("!")
\ No newline at end of file
+-- esperaIA::Integer -> () -- Sempre iniciar com 0
```

Comment:
Essa função não vai ser mais usada? Se não for, melhor remover do código pra n deixar poluído

Reply:
acredito que vai sim, mas como não é código de vitor nem meu, ficou comentado